source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -48,18 +63,12 @@ public class TentacleClient : ITentacleClient
             this.scriptObserverBackOffStrategy = scriptObserverBackOffStrategy;
             this.tentacleClientObserver = tentacleClientObserver;
 
-            var innerHandler = halibutRuntime.OverrideErrorResponseMessageCaching;
```

Comment:
Just to make sure I understand this correctly, it looks like every time this constructor is called, we take a reference to the existing inner `OverrideErrorResponseMessageCaching`, then wrap it another one.

Each time this is called, it adds another 'layer' to the onion, right?

And even though this lambda doesn't reference any properties of this class, it still results holds a reference to it, and that results in the memory leak you saw, right?

Reply:
Yep 🧅